gstreamer/src/pad.rs

```diff
@@ -183,7 +183,7 @@ pub trait PadExtManual {
 
     fn set_query_function<F>(&self, func: F)
     where
-        F: Fn(&Pad, &Option<::Object>, &mut ::QueryRef) -> bool + Send + Sync + 'static;
+        F: Fn(&Pad, &Option<::Object>, &mut ::Query) -> bool + Send + Sync + 'static;
```

Comment:
&mut Query has no meaning, as it's a refcount wrapper. You can always just clone it (increase refcount), and then you can get a mutable reference from it. Just that the inner type will have a refcount > 1 and is not mutable.

That's exactly the reason why before a &mut QueryRef was passed, a non-refcounted, borrowed reference, that statically has mutability guaranteed and can't in any case be cloned (or otherwise get the refcount increased).

In your design this should probably be a &mut QueryView?
Similar thoughts for Pad::query(), etc